metaflow/plugins/argo/argo_events_decorator.py

```diff
@@ -0,0 +1,154 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+# TODO: This decorator interface can be a top-level decorator to support similar
+#       implementations for Netflix Maestro, AWS Step Functions, Apache Airflow...
+class ArgoEventsDecorator(FlowDecorator):
+    name = "trigger"
+    defaults = {
+        "event": None,
+        "events": [],
+        "options": {},  # TODO: introduce support for options
+    }
+
+    def flow_init(
+        self, flow, graph, environment, flow_datastore, metadata, logger, echo, options
+    ):
+        self.triggers = []
+        if sum(map(bool, (self.attributes["event"], self.attributes["events"]))) > 1:
```

Comment:
Nit:

`if self.attributes["event"] and self.attributes["events']`

Also should we be checking with an XOR logic instead?  I.e. we want exactly one of the two to be defined right?

`if bool(self.attributes["event"]) == bool(self.attributes["events']):`

Reply:
Yes - you are correct. This is a remnant of when I had more options. I can simplify this.